src/service/xhr-impl.js

```diff
@@ -106,6 +113,15 @@ export class Xhr {
    * @private
    */
   fetch_(input, init) {
+    if (this.ampdoc_) {
+      // TODO(@jridgewell, #9048): We can alternatively _always_ wait on
+      // #whenFirstVisible.
```

Comment:
Could be nice as a conservative mitigation technique. As is, this PR won't prevent the next privacy issue.